Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/modules/tracing/DefaultTracingStatus.kt

```diff
@@ -84,23 +84,34 @@ class DefaultTracingStatus @Inject constructor(
             .addOnFailureListener { cont.resumeWithException(it) }
     }
 
+    @Suppress("LoopWithTooManyJumpStatements")
     override val isTracingEnabled: Flow<Boolean> = flow {
         while (true) {
             try {
                 emit(isEnabled())
                 delay(POLLING_DELAY_MS)
             } catch (e: CancellationException) {
-                Timber.d("isBackgroundRestricted was cancelled")
+                Timber.tag(TAG).d("isBackgroundRestricted was cancelled")
                 break
+            } catch (e: ApiException) {
+                emit(false)
+                if (e.statusCode == 17) {
+                    // No ENS installed, no need to keep polling.
+                    Timber.tag(TAG).v("No ENS available, aborting polling, assuming permanent.")
+                    break
+                } else {
+                    Timber.tag(TAG).v("Polling failed, will retry with backoff.")
+                    delay(POLLING_DELAY_MS * 5)
```

Comment:
does `retry` operator suits this case ? instead of `delay` ?

Reply:
I don't see a way to use the flow retry operator here without completely changing the polling logic here, this would be too much for the scope of the PR, but I'd happy to discuss changing it in another PR for 2.2 if you see potential.